mailbox/store/src/test/java/org/apache/james/mailbox/store/ThreadIdGuessingAlgorithmContract.java

```diff
@@ -202,4 +231,41 @@ void givenOldMailWhenAddNonRelatedMailsThenGuessingThreadIdShouldBasedOnGenerate
         assertThat(threadId.getBaseMessageId()).isEqualTo(newBasedMessageId);
     }
 
+    @Test
+    void givenThreeMailsInAThreadThenGetThreadShouldReturnAListWithMessageIdsInThatThread() throws MailboxException {
+        MailboxMessage message1 = createMessage(mailboxId, ThreadId.fromBaseMessageId(newBasedMessageId));
+        MailboxMessage message2 = createMessage(mailboxId, ThreadId.fromBaseMessageId(newBasedMessageId));
+        MailboxMessage message3 = createMessage(mailboxId, ThreadId.fromBaseMessageId(newBasedMessageId));
+
+        Mailbox mailbox = new Mailbox(MailboxPath.inbox(USER), UID_VALIDITY, mailboxId);
+        messageMapper.add(mailbox, message1);
+        messageMapper.add(mailbox, message2);
+        messageMapper.add(mailbox, message3);
+
+        Flux<MessageId> messageIds = testee.getMessageIdsInThread(ThreadId.fromBaseMessageId(newBasedMessageId), mailboxSession);
+
+        assertThat(messageIds.collectList().block()).isEqualTo(List.of(message1.getMessageId(), message2.getMessageId(), message3.getMessageId()));
+    }
+
+    @Test
+    void givenNonMailInAThreadThenGetThreadShouldThrowThreadNotFoundException() throws MailboxException {
+        Flux<MessageId> messageIds = testee.getMessageIdsInThread(ThreadId.fromBaseMessageId(newBasedMessageId), mailboxSession);
+
+        assertThatThrownBy(() -> testee.getMessageIdsInThread(ThreadId.fromBaseMessageId(newBasedMessageId), mailboxSession).collectList().block()).getCause().isInstanceOf(ThreadNotFoundException.class);
+    }
+
+    private SimpleMailboxMessage createMessage(MailboxId mailboxId, ThreadId threadId) {
+        MessageId messageId = mapperProvider.generateMessageId();
+        String content = "Some content";
+        int bodyStart = 16;
+        return new SimpleMailboxMessage(messageId,
+            threadId,
+            new Date(),
+            content.length(),
+            bodyStart,
+            new ByteContent(content.getBytes()),
+            new Flags(),
+            new PropertyBuilder().build(),
+            mailboxId);
+    }
 }
```

Comment:
One mail in a tread?

mail 1, mail 3 in thread 1
mail 2 in thread 2

get thread 1 -> mail 1 + mail 3 (not related emails are not returned)